### PR TITLE
Use Duration instead of int in ConfigurationProperties

### PR DESCRIPTION
**Pull request by jandusil**

### CODE REVIEW

**Comment by jandusil** [comment]
I am not sure about whether to create new migration file, or include it to PowerAuth-Server-1.5.0.md

File: docs/Configuration-Properties.md
Comment (banterCZ) [inline docs/Configuration-Properties.md:31]: May we use the variant without `PT` prefix? Spring Boot can do that, see https://docs.spring.io/spring-boot/docs/2.1.13.RELEASE/reference/html/boot-features-external-config.html#boot-features-external-config-conversion-duration

File: powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/configuration/PowerAuthServiceConfiguration.java
Comment (banterCZ) [inline powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/configuration/PowerAuthServiceConfiguration.java:226]: What is the difference between this and the previous format `@Min(0)`?

File: powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/ActivationServiceBehavior.java
Comment (banterCZ) [inline powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/ActivationServiceBehavior.java:690]: Are you sure that `timestamp` (before) is the same as `Instant.now()`? I think that there is a little time shift.

File: powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/CallbackUrlBehavior.java
Comment (banterCZ) [inline powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/CallbackUrlBehavior.java:434]: Not much dangerous here, but better to use `Math#toIntExact()`.

File: powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/configuration/PowerAuthServiceConfiguration.java
Comment (jandusil) [reply powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/configuration/PowerAuthServiceConfiguration.java:226]: The @Min apparently wont work on type Duration

File: docs/Configuration-Properties.md
Comment (jandusil) [reply docs/Configuration-Properties.md:31]: While it is true that we can omit the 'PT' prefix and still have Spring Boot interpret the duration correctly, I believe we should consider the implications for broader public documentation and usage. By adhering to the ISO 8601 Duration format in its complete form, including the 'PT' prefix, we ensure that our format is standard-compliant and immediately recognizable to anyone familiar with ISO 8601.

Comment (banterCZ) [reply docs/Configuration-Properties.md:31]: Standards are fine. Let's wait for another opinion and we will decide.

File: powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/CallbackUrlBehavior.java
Comment (banterCZ) [inline powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/CallbackUrlBehavior.java:434]: 🤔 I have opened another issue, what we may improve in the next step. https://github.com/wultra/lime-java-core/issues/201

File: powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/replay/ReplayPersistenceService.java
Comment (banterCZ) [inline powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/replay/ReplayPersistenceService.java:73]: Doesn't it work in a simpler form?
```suggestion
        final Instant expiration = Instant.now().plus(config.getRequestExpiration());
```

**Review by banterCZ — APPROVED** [review]
Good job (except my personal opinion about the duration format).
Let's wait for the fix of crypto 3.1 and run backend tests before merging.

**Pull request merged**
